crates/cadenza-eval/src/eval.rs

```diff
@@ -231,6 +231,38 @@ impl Eval for Ident {
 
 impl Eval for Apply {
     fn eval(&self, ctx: &mut EvalContext<'_>) -> Result<Value> {
+        // Check if this is a block expression (sequence of expressions)
+        // A block is when we have a receiver that is an Apply for a statement-like expression
+        // (like assignment with = or let) followed by arguments at this level.
+        // This pattern is created by indentation in the source code.
+        let receiver_opt = self.receiver().and_then(|r| r.value());
+        let local_args: Vec<Expr> = self.arguments().filter_map(|a| a.value()).collect();
+
+        if let Some(receiver_expr) = &receiver_opt {
+            if let Expr::Apply(receiver_apply) = receiver_expr {
+                // Check if the receiver Apply is a statement that returns nil
+                // (like let or = at the statement level)
+                if !local_args.is_empty() {
+                    if let Some(callee) = receiver_apply.callee() {
+                        if let Some(id) = extract_identifier(&callee) {
+                            let id_str: &str = &id;
+                            // Check if this is a statement-like form (let, =)
+                            // These return nil and shouldn't be called with arguments
+                            if id_str == "let" || id_str == "=" {
```

Comment:
all of this is incredibly specific and brittle. what if i wanted to define a function? what if i'm calling an arbitrary function? I would expect to be able to iterate over each child in a block and evaluate it, similar to how we do for the root. and then you pop a scope once you exit.

that being said, i think block should be a macro where the callee is a synthetic block. and then we'd call that macro, which would then evaluate each item and return the last one. so we will likely need to fix the parser to insert that.